astropy/io/fits/tests/test_image.py

```diff
@@ -168,6 +168,45 @@ def test_open_3(self):
         finally:
             fits.conf.lazy_load_hdus = True
 
+    def test_io_CvsF():
+        # Test that files are being correctly written+read for "C" and "F" order arrays 
+        dims = (3,7)
+        size = dims[0] * dims[1]
+        a = np.zeros(dims, order='C').reshape(*dims)
+        b = np.zeros(dims, order='F').reshape(*dims)
```

Comment:
The `reshape` on `np.zeros` arrays is useless I think.

A few comments for the following:

- I don't think that pytest will test the nested functions. You need to make them proper methods, passing self.
- The pathlib test is useless I think, it will be converted to a string so it is the same as the first one.
- Then you could merge the two test functions in `test_fortran_array` or something similar.
- Use `np.assert_array_equal` instead of `np.all`, no need to test the shape separately.

Reply:
* you're right on nested test functions, I should have seen that
* I removed the pathlib test
* merging done
* I did not know that function (part of the `testing` submmodule btw), thanks !

Reply:
I'll get back to the reshape when we get a satisfactory fix :)